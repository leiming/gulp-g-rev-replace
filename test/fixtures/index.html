<!DOCTYPE html>
<html>
<head lang="en">
  <meta charset="UTF-8">
  <title>Test</title>
  <!-- build:css -->
  <link rel="stylesheet" href="../reboot/bundle.css"/>
  <!-- endbuild -->
</head>
<body>
<script src="http://libs.useso.com/js/jquery/1.9.1/jquery.min.js"></script>
<script src="http://js.passport.qihucdn.com/5.0.2.js"></script>
<!-- build:js -->
<script src="http://u.youxi.com/js/mylogin/yuc2.0.js"></script>
<script src="../reboot/harry.js"></script>
<script src="../reboot/bar/foo.js?aaa=123&bbb=ccc"></script>
<script src="../reboot/bar/jam.js?aaa=123&bbb=ccc"></script>
<!-- endbuild -->
</body>
</html>