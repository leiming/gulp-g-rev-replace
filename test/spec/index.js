"use strict";
/* global describe, it, expect */

var fs = require('fs')
var path = require('path')
var gutil = require('gulp-util')
var rev_replace = require('../../index')

function getFile(filePath) {
  return new gutil.File({
    path:     filePath,
    cwd:      __dirname,
    base:     path.dirname(filePath),
    contents: fs.readFileSync(filePath)
  });
}

function getFixture(filePath) {
  return getFile(path.join('test', 'fixtures', filePath))
}

function getExpected(filePath) {
  return getFile(path.join('test', 'expected', filePath))
}

function compare(name, expectedName, done) {
  var stream = rev_replace();

  stream.on('data', function(newFile) {
    if (path.basename(newFile.path) === name) {
      expect(String(getExpected(expectedName).contents)).toEqual(String(newFile.contents))
    }
  });

  stream.on('end', function() {
    done();
  });

  stream.write(getFixture(name))

  stream.end()
}

describe('hello', function(){
  it('good', function () {
    expect('bar').toBe('bar')
  })
})

