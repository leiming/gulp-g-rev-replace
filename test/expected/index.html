<!DOCTYPE html>
<html>
<head lang="en">
  <meta charset="UTF-8">
  <title>Test</title>
    <link rel="stylesheet" href="http://s0.qhimg.com/default/bundle/0_0_4.css"/>
  </head>
<body>
<script src="http://libs.useso.com/js/jquery/1.9.1/jquery.min.js"></script>
<script src="http://js.passport.qihucdn.com/5.0.2.js"></script>
<script src="http://u.youxi.com/js/mylogin/yuc2.0.js"></script>
<script src="http://s0.qhimg.com/default/harry/0_0_4.js"></script>
<script src="http://s0.qhimg.com/default/bar;foo/0_0_4.js?aaa=123&bbb=ccc"></script>
<script src="http://s0.qhimg.com/default/bar;jam/0_0_4.js?aaa=123&bbb=ccc"></script>
</body>
</html>